pkg/sca/testdata/shebang-test.yaml

```diff
@@ -0,0 +1,47 @@
+package:
+  name: shebang-test
+  version: 1
+  epoch: 0
+  description: shebang test
+  copyright:
+    - license: MIT
+
+environment:
+  contents:
+    packages:
+      - busybox
+
+pipeline:
+  - runs: |
+      wbin() {
+          local f="$1"
+          cat > "$BD/usr/bin/$f"
+          echo "wrote usr/bin/$f" 1>&2
+          chmod 755 "$BD/usr/bin/$f"
+      }
+     
+      BD=${{targets.destdir}}
+      mkdir -p "$BD/usr/bin"
+     
+      wbin "bash-via-env" <<"EOF"
+      #!/usr/bin/env bash
+      echo hi world
+      EOF
+      
+      wbin "bash-straight-up" <<"EOF"
+      #!/bin/bash
+      echo hi from bash
+      EOF
+      
+      wbin "python-via-env" <<"EOF"
+      #!/usr/bin/env python3
+      print("hello world")
+      EOF
+      
+      wbin "python-straight" <<"EOF"
+      #!/usr/bin/python3
```

Comment:
hm..... I hoped we generate most of our scripts with python3.12 shebangs.
Or if we do have scripts with python3 shebang + modules => we need a dep on python-3.12 (which would allow executing the module, and has the python3 symlink).

Reply:
I think we do generate "most of our scripts" with usr/bin/python3.12 shebang , and we absolutely should.

The change here adds a 'cmd:<thing>' for the thing in the shebang.  So that should work with 'python3.12', or 'python3' or 'bash', as the package that has that program will provide the 'cmd:thing' correctly.  Am I missing something there?